### PR TITLE
[JENKINS-55848] Identify the name of the thing being deleted

### PR DESCRIPTION
**Pull request by jsoref**
Note: this is especially helpful on mobile where the message is very far removed from the top which may have the name...

See [JENKINS-55848](https://issues.jenkins-ci.org/browse/JENKINS-55848).

### Proposed changelog entries

* Report the name of the item being deleted

### Submitter checklist

- [x] JIRA issue is well described
- [x] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [n/a] Appropriate autotests or explanation to why this change has no tests
- [n/a] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@daniel-beck this is what I was offering to do a few days ago...

#### Improvements
* image
* image
* image
* image

#### Odd divergence
* image

This item could be changed to not specify the name, but, it's odd that it used `Delete this build` instead of just `Delete build`, so I've left the change in to show that it sticks out like 

#### Atypical rendering
* image

#### Not fixed
image

### CODE REVIEW

File: core/src/main/resources/hudson/logging/LogRecorder/delete.jelly [hunk 1]
```diff
@@ -31,9 +31,9 @@ THE SOFTWARE.
     <st:include page="sidepanel.jelly" />
     <l:main-panel>
       <form method="post" action="doDelete">
-        ${%Are you sure about deleting this log recorder?}
+        ${%delete.logrecorder(it.displayName)}
```
Comment (Wadeck) [inline core/src/main/resources/hudson/logging/LogRecorder/delete.jelly]: Unfortunately there are ~10 translations with the "very nice" key: `Are\ you\ sure\ about\ deleting\ this\ log\ recorder?`. So this improvement for english will result in quality decrease for all the other languages.

**Review by Wadeck — COMMENTED** [review]
This initiative seems interesting. I think the drawback could be reduce drastically by involving people from different languages contributing with their respective translation.

As a possible "temporary" solution, you can propose something like:
- in the different languages, you change the translation key to the new one
- adding a simple `[...] : {1}` at the end, as a simple implementation
- that would be wrong for some languages of course, but at least we can go on

WDYT?

**Comment by jsoref** [comment]
@Wadeck: here's a draft of that done using some random code:
```
hg chistedit '. % master'
```
#### loop
```
hg diff -U0|perl -ne 'next unless /^- /;s/.*\%//;s/[}"].*//;s/ /\\\\ /g;chomp;$y=$_; $y=~s/\?/./; print "hg locate -0 #*_*.properties#|xargs -0 grep -l #$_#|xargs perl -pi -e #next unless /$y/;s/\\?\$/\? ({0})\n/;#;"' |tr "#" "'" > ~/pass1; sh -x ~/pass1
```
(there could have been code to calculate the delete variable, but it wasn't worth it):
```
d -U0 > ~/pass2; patch -R -p1 < ~/pass2; export delete=delete.user; perl -pi.a -e 'next unless s/^\+.*\?=/+'$delete'=/;' 0; patch -p1 < ~/pass2
EDITOR=true hg histedit --cont
```
#### end loop

**Review by Wadeck — APPROVED** [review]
That seems a lot better! No drawback for other languages, an additional information "as bonus" and the full improvement for english.

Thank you for the update!

**Review by oleg-nenashev — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Thanks @jsoref !

**Comment by mmh891113** [comment]
sometimes  displayName is too long， how about this issue： https://issues.jenkins-ci.org/browse/JENKINS-58157

File: core/src/main/resources/hudson/logging/LogRecorder/delete.jelly (hunk 1 above)
Comment (mmh891113) [reply core/src/main/resources/hudson/logging/LogRecorder/delete.jelly]: https://issues.jenkins-ci.org/browse/JENKINS-58157

**Comment by jsoref** [comment]
Commenting twice won't get it fixed faster...

I suppose I could add some CSS that either force wraps or clips. If I go with clipping, I could have it unclip on hover (and probably z- to front).

**Pull request merged**